### ISSUE TITLE
Flow of a single element doesn't throw error when the element doesn't exist

### ISSUE THREAD

**Issue by tamimattafi** [tamimattafi/kabin#3]
## Problem
Querying a reactive entity doesn't throw an exception, if the entity doesn't exist, since `Dao` is using `mapToOneNotNull` instead of `mapToOne`

However, **Room** uses a different behavior, it emits an error instead of suspending the `Flow<T>` forever while waiting for insert

## Example
Here's a query that returns a Flow of a single element:
```kotlin
@Query("SELECT * FROM UserEntity WHERE name = :name AND sampleAge = :age")
suspend fun getEntityReactive(age: Int, name: String): Flow<UserEntity>
```

This will generate the following code:
```kotlin
override suspend fun getEntityReactive(age: Int, name: String): Flow<UserEntity> =
    queries.queryUserEntityByNameAge(name, age).asFlowIONotNull()
```

As we see, converting `Query<T>` to `Flow<T>` is done using `asFlowIONotNull`, but it will be more correct to use `asFlowIO`, which uses `mapToOne` under the hood

**Comment by tamimattafi** [comment]
`asFlowIO` will throw `NullPointerException: ResultSet returned null for query` if the element doesn't exist

**Comment by tamimattafi** [comment]
Testing in a real environment resulted in a success, therefore, this issue is considered solved.
Changes were published to [0.1.0-alpha02](https://github.com/tamimattafi/kabin/releases/tag/0.1.0-alpha02)
